packages/blocks/code/src/App.tsx

```diff
@@ -86,12 +87,29 @@ export const App: BlockComponent<AppProps> = ({
     }
   };
 
-  const handleCaptionBtnClick = () => {
-    if (!captionRef.current) return;
-    if (captionRef.current.classList.contains("invisible")) {
-      captionRef.current.classList.remove("invisible");
+  const handleCaptionButtonClick = () => {
+    setCaptionVisibility(true);
+    setTimeout(() => {
+      captionRef.current?.focus();
+      captionRef.current?.setSelectionRange(
+        0,
+        captionRef.current?.value.length,
+      );
+    }, 0);
+  };
+
+  useEffect(() => {
+    if (captionRef.current !== document.activeElement) {
```

Comment:
In the future, we should consider tracking whether the caption is focused or not in state, so that this doesn't need to be done in an effect